config/testdata/conf.good.yml

```diff
@@ -109,7 +109,8 @@ scrape_configs:
 - job_name: service-kubernetes
 
   kubernetes_sd_configs:
-  - api_servers:
+  - kind: service_endpoints
```

Comment:
Sorry I think I missed the detail of that discussion. Pod discovery is the only way for us to discover actually running stuff. Other abstractions, e.g. service/daemonset/job/etc, only allow us to discover `Ready` pods & it was discussed a while back that scraping targets that are starting up is desirable for a number of reasons. Only pod discovery can give us that. The one thing that it doesn't give us is a useful `job` label, which is where I think we're sadly going to have to re-implement the Kubernetes selector logic to give us that `job` label, unless someone can think of a better approach.